backend/seeder/seeder.py

```diff
@@ -0,0 +1,236 @@
+
+
+def student_generator():
+    """Generates a student user object"""
+    return user_generator('STUDENT')
+
+
+def admin_generator():
+    """Generates an admin user object"""
+    return user_generator('ADMIN')
+
+
+def user_generator(role):
+    """Generates a user object with the given role"""
+    user = User(uid=generate_random_uid(),
+                role=role,
+                display_name=fake.name())
+    return user
+
+
+def course_student_generator(course_id, uid):
+    """Generates a course student relation object"""
+    return CourseStudent(course_id=course_id, uid=uid)
+
+
+def course_admin_generator(course_id, uid):
+    """Generates a course admin relation object"""
+    return CourseAdmin(course_id=course_id, uid=uid)
+
+
+def generate_course(teacher_uid):
+    """Generates a course object with a random name and the given teacher uid"""
+    course = Course(name=generate_course_name(),
+                    teacher=teacher_uid)
+    return course
+
+
+def generate_projects(course_id, num_projects):
+    """Generates a list of project objects with random future deadlines"""
+    projects = []
+    for _ in range(num_projects):
+        deadlines = []
+        # Generate a random number of deadlines (1 to 3)
+        num_deadlines = random.randint(1, 3)
+
+        for _ in range(num_deadlines):
+            future_datetime = datetime.now() + timedelta(days=random.randint(1, 30))
+            deadline = (fake.catch_phrase(), future_datetime)
+            deadlines.append(deadline)
+        project = Project(
+            title=fake.catch_phrase(),
+            description=fake.catch_phrase(),
+            deadlines=deadlines,
+            course_id=course_id,
+            visible_for_students=random.choice([True, False]),
+            archived=random.choice([True, False]),
+            regex_expressions=[]
+        )
+        projects.append(project)
+    return projects
+
+
+def generate_submissions(project_id, student_uid):
+    """Generates a list of submissions with random status"""
+    submissions = []
+    statusses = [SubmissionStatus.SUCCESS, SubmissionStatus.FAIL,
+                SubmissionStatus.LATE, SubmissionStatus.RUNNING]
+    num_submissions = random.randint(0, 2)
+    for _ in range(num_submissions):
+        submission = Submission(project_id=project_id,
+                                uid=student_uid,
+                                submission_time=datetime.now(),
+                                submission_path="",
+                                submission_status=
+                                random.choice(statusses))
+        graded = random.choice([True, False])
+        if graded and submission.submission_status == "SUCCESS":
+            submission.grading = random.randint(0, 20)
+        submissions.append(submission)
+    return submissions
+
+
+def into_the_db(my_uid):
+    """Populates the db with 5 courses where my uid is teacher and 5 where he is student"""
```

Comment:
Make this clearer by using my_uid instead of my uid, because the argument was always referred to as my_uid before